core/commands/get.go

```diff
@@ -251,15 +255,15 @@ func (gw *getWriter) writeExtracted(r io.Reader, fpath string) error {
 	return extractor.Extract(r)
 }
 
-func getCompressOptions(req cmds.Request) (int, error) {
-	cmprs, _, _ := req.Option("compress").Bool()
-	cmplvl, cmplvlFound, _ := req.Option("compression-level").Int()
+func getCompressOptions(req *cmds.Request) (int, error) {
+	cmprs, _ := req.Options["compress"].(bool)
+	cmplvl, _ := req.Options["compression-level"].(int)
 	switch {
 	case !cmprs:
 		return gzip.NoCompression, nil
-	case cmprs && !cmplvlFound:
+	case cmprs && cmplvl == -1:
```

Comment:
Why? Why not just use `cmplvlFound`? We should return an error if is `cmplvl < 0`.